uxarray/grid/coordinates.py

```diff
@@ -227,9 +115,86 @@ def _populate_face_centroids(grid, repopulate=False):
         )
 
 
+def _populate_face_centerpoints(grid, repopulate=False):
```

Comment:
Can we add a benchmark to our `mpas_ocean` suite that benchmarks the execution time for this function across the 480km and 120km resolutions? 

Something like what @aaronzedwick put together in #870

Reply:
It is slow, looking to see optimization possibilities.

Reply:
> It is slow, looking to see optimization possibilities.

was able to fix Numba causes issues with recursive functions, this is much more efficient ~5-10x faster.